### PR TITLE
`WPS525` forbids comparisons where `in` is compared with single item container

### LINKED ISSUE #921: Forbid single item in "in" operators
# Rule request

## Thesis

These two code samples are identical:

```python
a in {'s'}
```

and 

```python
a == 's'
```

But, the second version is more clear. We should enforce it.

## Reasoning

"in" operator with single item look like overhead. Let's do things more clear!

### PR DESCRIPTION
**Pull request by slapshin**
## Checklist

- [x] I have double-checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [x] I have created at least one test case for the changes I have made
- [x] I have updated the documentation for the changes I have made
- [x] I have added my changes to the `CHANGELOG.md`

## Related issues

Format is:
- Closes #921 

I have isolated logic from `CompareSanityVisitor` related to checking `in` expressions into `InCompareSanityVisitor`. Reason: when I had added a new rule, `CompareSanityVisitor` already was contained many logic and methods. Flake8 forbade it. So I have created new visitor `InCompareSanityVisitor` and moved `MultipleInCompareViolation`, `WrongInCompareTypeViolation` and `InCompareWithSingleItemContainerViolation` to it. Please check it. Am I right or not?

### CODE REVIEW

File: tests/test_visitors/test_ast/test_compares/test_in_with_single_item_container.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    InCompareWithSingleItemContainerViolation,
+    WrongInCompareTypeViolation,
+)
+from wemake_python_styleguide.visitors.ast.compares import (
+    InCompareSanityVisitor,
+)
+
+
+@pytest.mark.parametrize('code', [
+    'if a in {1}: ...',
```
Comment (sobolevn) [inline tests/test_visitors/test_ast/test_compares/test_in_with_single_item_container.py]: We also need to check dicts: `{1: 'a'}`

File: tests/test_visitors/test_ast/test_compares/test_in_with_single_item_container.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    assert_errors(
+        visitor,
+        [InCompareWithSingleItemContainerViolation],
+        ignored_types=(WrongInCompareTypeViolation,),
+    )
+
+
+@pytest.mark.parametrize('code', [
+    'if a in {1, 2}: ...',
```
Comment (sobolevn) [inline tests/test_visitors/test_ast/test_compares/test_in_with_single_item_container.py]: The same here.

File: wemake_python_styleguide/violations/refactoring.py [hunk 3]
```diff
@@ -982,3 +984,34 @@ class MisrefactoredAssignmentViolation(ASTViolation):
...
+    Solution:
+        Refactor your code to use ``==`` instead ``in``.
+
+    Example::
+
+        # Correct:
+        a == 's'
+
+        # Wrong:
+        a in {'s'}
```
Comment (sobolevn) [inline wemake_python_styleguide/violations/refactoring.py]: `a in {'s'}` is enough, others are just duplicates

File: wemake_python_styleguide/visitors/ast/compares.py [hunk 4]
```diff
@@ -416,3 +391,61 @@ def _check_incorrect_not(self, node: ast.UnaryOp) -> None:
 
         if isinstance(node.operand, ast.Compare):
             self.add_violation(NotOperatorWithCompareViolation(node))
+
+
+@final
+class InCompareSanityVisitor(BaseNodeVisitor):
+    """Restricts the incorrect ``in`` compares."""
+
+    in_nodes: ClassVar[AnyNodes] = (
```
Comment (sobolevn) [inline wemake_python_styleguide/visitors/ast/compares.py]: `_in_nodes` - should be protected

Comment (slapshin) [reply wemake_python_styleguide/visitors/ast/compares.py]: sorry, my mistake

**Pull request merged**